contest/contest_info.py

```diff
@@ -107,7 +111,74 @@ def get_scoreboard(contest):
 
     return scoreboard
 
+def get_scoreboard_csv(contest_id, scoreboard_type):
+    contest = get_contest_or_404(contest_id)
+    scoreboard = get_scoreboard(contest)
+    
+    response = HttpResponse(content_type='text/csv')
+    filename = str(contest.cname) + '-scoreboard-' + str(scoreboard_type)
+    response['Content-Disposition'] = 'attachment; filename=' + filename
+
+    #init
+    writer = csv.writer(response)
+    if scoreboard_type == "penalty":
+        write_scoreboard_csv_penalty(writer, contest, scoreboard)
+    elif scoreboard_type == "testcases":
+        write_scoreboard_csv_testcases(writer, contest, scoreboard)
+
+    return response
+
+def write_scoreboard_csv_penalty(writer, contest, scoreboard):
+    #penalty scoreboard csv
+    scoreboard.sort_users_by_penalty()
+    #title
+    title = ['Penalty', 'User']
+    for problem in scoreboard.problems:
+        title.append(problem.id)
+    title.append('Total')
+    writer.writerow(title)
+    #user data
+    for counter, user in enumerate(scoreboard.users):
+        user_row = [counter+1, user.username]
+        for problem in user.problems:
+            penalty = problem.get_penalty(contest.start_time)
+            user_row.append(penalty)
+        total_penalty = user.get_penalty(contest.start_time)
+        user_row.append(total_penalty)
```

Comment:
I prefer using another format:
```submissions before first AC``` / ```first AC time```(in minutes)
if a problem didn't get AC, use ```--``` instead
For example,
if an user got AC for the first submission, and submission time is 10 minutes after contest start, use ```1/10```
if an user had tried 4 times but never got AC, use ```4/--```